app/src/main/java/com/example/playlistmaker/activities/SearchActivity.kt

```diff
@@ -140,19 +148,24 @@ class SearchActivity : AppCompatActivity() {
         persistentState: PersistableBundle?
     ) {
         super.onRestoreInstanceState(savedInstanceState, persistentState)
-        searchText = savedInstanceState?.getString(searchField, searchFieldDefault) ?: ""
+        searchText = savedInstanceState?.getString(SEARCH_FIELD, SEARCH_FIELD_DEFAULT) ?: ""
         binding.etSearch.setText(searchText)
     }
 
     override fun onSaveInstanceState(outState: Bundle, outPersistentState: PersistableBundle) {
         super.onSaveInstanceState(outState, outPersistentState)
-        outState.putString(searchField, searchText)
+        outState.putString(SEARCH_FIELD, searchText)
     }
 
     private fun search(text: String) {
+        binding.progressBar.visibility = View.VISIBLE
+        binding.rvTracksSearch.visibility = View.GONE
+        binding.lrPlaceHolder.visibility = View.GONE
         iTunesService.findTrack(text).enqueue(object : Callback<TracksResponse> {
             @SuppressLint("NotifyDataSetChanged")
             override fun onResponse(call: Call<TracksResponse>, response: Response<TracksResponse>) {
+                binding.progressBar.visibility = View.GONE
+                binding.rvTracksSearch.visibility = View.VISIBLE
                 if (response.code() == 200) {
```

Comment:
Чтобы не хардкодить коды http запросов можно воспользоваться проперти `isSuccessfull`:
```suggestion
                if (response.isSuccessfull) {
```